server/planning/planning/planning_spike_async.py

```diff
@@ -0,0 +1,192 @@
+from copy import deepcopy
+from typing import Any
+
+from planning.types.assignment import AssignmentResourceModel
+from planning.types.event import EventResourceModel
+from planning.types.planning import PlanningResourceModel
+from superdesk.resource_fields import ID_FIELD
+from superdesk import get_resource_service
+from superdesk.notification import push_notification
+from superdesk.errors import SuperdeskApiError
+from apps.auth import get_user, get_user_id
+from apps.archive.common import get_auth
+
+from planning.utils import get_related_event_ids_for_planning, get_first_related_event_id_for_planning
+from planning.common import (
+    ITEM_EXPIRY,
+    ITEM_STATE,
+    set_item_expiry,
+    WORKFLOW_STATE,
+    get_coverage_type_name,
+    remove_autosave_on_spike,
+    remove_lock_information,
+)
+from planning.planning_notifications import PlanningNotifications
+from planning.item_lock import LOCK_USER
+
+
+def post_update_planning_item_actions(updates: dict[str, Any], original: dict[str, Any]):
+    if original.get(LOCK_USER) and LOCK_USER in updates and updates[LOCK_USER] is None:
+        push_notification(
+            "planning:unlock",
+            item=str(original.get(ID_FIELD)),
+            user=str(get_user_id()),
+            lock_session=str(get_auth().get(ID_FIELD)),
+            etag=updates.get("_etag"),
+            event_ids=get_related_event_ids_for_planning(original),
+            recurrence_id=original.get("recurrence_id") or None,
+            type=original.get("type"),
+        )
+
+
+async def post_planning_item_spike_actions(updates: dict[str, Any], original: dict[str, Any]):
+    post_update_planning_item_actions(updates, original)
+    events_service = EventResourceModel.get_service()
+
+    # Delete assignments in workflow
+    assignments_to_delete = []
+    coverages = original.get("coverages") or []
+    for coverage in coverages:
+        if coverage.get("workflow_status") == WORKFLOW_STATE.ACTIVE:
+            assignments_to_delete.append(coverage)
+
+    notify_user_on_failed_assignment_deletes = True
+    first_event_id = get_first_related_event_id_for_planning(original, "primary")
+
+    if first_event_id:
+        event = await events_service.find_by_id_raw(first_event_id)
+        notify_user_on_failed_assignment_deletes = not event or event.get("state") != WORKFLOW_STATE.SPIKED
+
+    # TODO-ASNYC: Convert this to async call when function is added to PlanningAsyncService
+    get_resource_service("planning").delete_assignments_for_coverages(
```

Comment:
This looks like a small enough function, would you be able to re-create this function in async in the `planning_utils` file in this PR and use it here?